src/main/java/org/ibp/api/java/impl/middleware/dataset/DatasetServiceImpl.java

```diff
@@ -949,4 +956,27 @@ private void processSearchComposite(final SearchCompositeDto<ObservationUnitsSea
 			searchDTO.setSearchRequest(searchRequest);
 		}
 	}
+
+	private void correctKSUDateFormatIfNecessary(final Table<String, String, String> table, final List<MeasurementVariable> measurementVariables) {
+		final List<String> dateMeasurementVariables = measurementVariables.stream().filter(measurementVariable -> measurementVariable.getDataTypeId() == TermId.DATE_VARIABLE
+			.getId()).collect(Collectors.toList()).stream().map(measurementVariable -> measurementVariable.getName()).collect(Collectors.toList());
+		if (dateMeasurementVariables != null) {
+			for (final String obsUnit : table.rowKeySet()) {
+				for (final String colVariable : table.columnKeySet()) {
+					if (dateMeasurementVariables.contains(colVariable)) {
+						String value = table.get(obsUnit, colVariable);
+						final Date parsed = Util.tryParseDateAccurately(value, Util.DATE_AS_NUMBER_FORMAT);
```

Comment:
Hi @gelli-leafnode  , this line and go straight to checking if the value is in KSU format. If it is then proceed to changing the value to the valid format.